main/src/library/File.flix

```diff
@@ -520,37 +554,41 @@ namespace File {
     /// Returns an iterator of the bytes in the given `file` in chunks of size `chunkSize`.
     ///
     pub def readChunks(f: String, chunkSize: Int32): Iterator[Result[Array[Int8, Static], String], Static] & Impure =
-        try {
-            import new java.io.File(String): ##java.io.File & Impure as newFile;
-            import new java.io.FileInputStream(##java.io.File): ##java.io.FileInputStream & Impure as newInputStream;
-            import java.io.FileInputStream.read(Array[Int8, Static]): Int32 & Impure;
-
-            let javaFile = newFile(f);
-            let stream = newInputStream(javaFile);
-            let bytes = ref [0i8; chunkSize];
-            let numberRead = ref read(stream, deref bytes);
-
-            let done = () -> deref numberRead == -1;
-            let next = () -> {
-                try {
-                    let chunk = deref bytes;
-                    bytes := [0i8; chunkSize];
-                    let accRead = deref numberRead;
-                    numberRead := read(stream, deref bytes);
-                    Ok(readBytesWithHelper(chunk, accRead, chunkSize))
-                } catch {
-                    case ex: ##java.io.IOException => {
-                        numberRead := -1;
-                        import java.lang.Throwable.getMessage(): String & Impure;
-                        Err(getMessage(ex))
+        if (0 < chunkSize) {
```

Comment:
We should force the user to not use chunks of size 0, right?

Reply:
I think an empty iterator should be returned

Reply:
What do you mean with empty? The return type still needs to be `Iterator[Result[Array[Int8, Static], String], Static]`. My solution was :
```
Iterator.singleton(Static, Err("The argument `chunkSize` has to be bigger than 0 when calling `File.readChunks`."))
```
but I'm not sure this is what you mean by empty.

Reply:
`Iterator.new(r)` such that it is instantly empty instead of an error element since nothing went wrong there is just no information